recipes/glog/all/conanfile.py

```diff
@@ -18,83 +21,88 @@ class GlogConan(ConanFile):
         "fPIC": [True, False],
         "with_gflags": [True, False],
         "with_threads": [True, False],
+        "with_unwind": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
         "with_gflags": True,
         "with_threads": True,
+        "with_unwind": True,
     }
 
-    exports_sources = ["CMakeLists.txt", "patches/**"]
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            del self.options.with_unwind
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
         if self.options.with_gflags:
             self.options["gflags"].shared = self.options.shared
 
+    def layout(self):
+        cmake_layout(self)
+
     def requirements(self):
         if self.options.with_gflags:
             self.requires("gflags/2.2.2")
+        if self.options.get_safe("with_unwind"):
+            self.requires("libunwind/1.6.2")
 
     def build_requirements(self):
-        if tools.Version(self.version) >= "0.6.0":
+        if Version(self.version) >= "0.6.0":
             self.build_requires("cmake/3.22.3")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["WITH_GFLAGS"] = self.options.with_gflags
+        tc.variables["WITH_THREADS"] = self.options.with_threads
+        tc.variables["BUILD_SHARED_LIBS"] = self.options.shared
+        if Version(self.version) >= "0.5.0":
+            tc.variables["WITH_PKGCONFIG"] = True
+            if self.settings.os == "Emscripten":
+                tc.variables["WITH_SYMBOLIZE"] = False
+                tc.variables["HAVE_SYSCALL_H"] = False
+                tc.variables["HAVE_SYS_SYSCALL_H"] = False
+            else:
+                tc.variables["WITH_SYMBOLIZE"] = True
+            tc.variables["WITH_UNWIND"] = self.options.get_safe("with_unwind", default=False)
```

Comment:
What happen before 0.5.0 regarding libunwind?

Reply:
I think it was never used before, at least glog didnt have the `FindUnwind.cmake` so it probably just never found it. Just recently, starting with 0.5.0 it looked for it and this caused problems

Reply:
So libunwind should be added to requirements only if version >= 0.5.0 and (Linux or FreeBSD).

Reply:
Nop: it was unconditionally added: https://github.com/google/glog/blob/v0.4.0/CMakeLists.txt#L122-L123.
So:
- `with_unwind` should not exist before 0.5.0
- libunwind should be in requirements if version < 0.5.0 or (unwind True and (Linux or FreeBSD))

Reply:
Well, if unwind it was not found, the library was still compiled, just without the definition `HAVE_LIBUNWIND_H` (thus implicitly disabled). But it could still pick up system headers.